www/js/survey/inputMatcher.ts

```diff
@@ -295,36 +292,37 @@ export function mapInputsToTimelineEntries(
     } else {
       // MULTILABEL configuration: use the label inputs from the labelOptions to determine which
       // keys to look for in the unprocessedInputs
-      const labelsForTrip: { [k: string]: LabelOption } = {};
+      const labelsForTrip: { [k: string]: UserInputEntry | undefined } = {};
       Object.keys(getLabelInputDetails()).forEach((label: MultilabelKey) => {
         // Check unprocessed labels first since they are more recent
         const userInputForTrip = getUserInputForTimelineEntry(
           tlEntry,
           nextEntry,
           unprocessedLabels[label],
-        );
+        ) as UserInputEntry;
         if (userInputForTrip) {
-          labelsForTrip[label] = labelOptionByValue(userInputForTrip.data.label, label);
+          labelsForTrip[label] = userInputForTrip;
         } else {
           const processedLabelValue = tlEntry.user_input?.[inputType2retKey(label)];
-          labelsForTrip[label] = labelOptionByValue(processedLabelValue, label);
+          if (processedLabelValue) {
+            // TODO: when we unify the user input types on the server, we can remove this 'any' cast
+            labelsForTrip[label] = { data: { label: processedLabelValue } } as any;
+          }
         }
       });
       if (Object.keys(labelsForTrip).length) {
         timelineLabelMap[tlEntry._id.$oid] = labelsForTrip;
       }
     }
-  });
 
-  if (
-    appConfig?.survey_info?.buttons?.['trip-notes'] ||
-    appConfig?.survey_info?.buttons?.['place-notes']
-  ) {
-    // trip-level or place-level notes are configured, so we need to match additions too
-    allEntries.forEach((tlEntry, i) => {
```

Comment:
This change was so that instead of looping through `allEntries` twice (once for user inputs, once for additions), it loops once and does the `if` checks inside.

I did that while unifying the user input structure but probably forgot or neglected to leave a comment